download_logs.py

```diff
@@ -20,41 +23,101 @@ def get_arguments():
+    parser.add_argument('--mav_type', type=str, default=None, nargs = '+',
+                        help='Which mav types to download. Multiple are possible. Example: Quadrotor') # 1 +
+    parser.add_argument('--flight_modes', type=str, default=None, nargs='+',
+                        help='Whether to download only data with special flight modes '
+                             'Example: Mission')
+    parser.add_argument('--error_labels', default=None, nargs='+', type=str,
+                        help='Whether to only download data with special error labels '
+                             ' Example: Vibration')
+    parser.add_argument('--rating', default=None,  type=str, nargs='+',
+                        help='possible ratings of the data. Example: good excellent')
     return parser.parse_args()
 
 
+def flight_modes_to_ids(flight_modes):  # gets a string 'Manual' id=Zahl
+    """
+    returns a list of mode ids for a list of mode labels
+    """
+    flight_ids = []
+    for i in flight_modes_table:
+        if flight_modes_table[i][0] in flight_modes:
+            flight_ids.append(i)  # adds to list
+    return flight_ids
+
+
+def error_labels_to_ids(error_labels):  # gets a string
+    """
+    returns a list of error ids for a list of error labels
+    """
+    error_id_table = {label: id for id, label in error_labels_table.items()} # creates new dictionary
+    error_ids = [error_id_table[error_label] for error_label in error_labels] # creates list of error_ids
+    return error_ids
+
+
 def main():
     args = get_arguments()
 
     try:
         # the db_info_api sends a json file with a list of all public database entries
-        db_entries_list = requests.get(url=args.db_info_api).json()
+        db_entries_list = requests.get(url=args.db_info_api).json() # copy list
     except:
         print("Server request failed.")
         raise
 
     if args.print_entries:
-        # only print the json output
+        # only print the json output without downloading
         print(db_entries_list)
+
     else:
-        if not os.path.isdir(args.download_folder):
+        if not os.path.isdir(args.download_folder): # returns true if path is an existing directory
             print("creating download directory " + args.download_folder)
-            os.makedirs(args.download_folder)
+            os.makedirs(args.download_folder) # creates new folder
         # find already existing logs in download folder
         logfile_pattern = os.path.join(os.path.abspath(args.download_folder), "*.ulg")
         logfiles = glob.glob(os.path.join(os.getcwd(), logfile_pattern))
         logids = frozenset(os.path.splitext(os.path.basename(f))[0] for f in logfiles)
 
+        # filter for mav types
+        if args.mav_type is not None:
+            mav = [mav_type.lower() for mav_type in args.mav_type]
+            db_entries_list = [entry for entry in db_entries_list if set([entry["mav_type"].lower()]).issubset(set(mav))]
```

Comment:
Instead of `set([entry["mav_type"].lower()]).issubset(set(mav))`, why not just `entry["mav_type"].lower() in mav`?